rln-wasm/src/lib.rs

```diff
@@ -20,6 +22,45 @@ pub struct RLNWrapper {
     instance: RLN<'static>,
 }
 
+// Macro to call methods with arbitrary amount of arguments,
+// which have the last argument is output buffer pointer
+// First argument to the macro is context,
+// second is the actual method on `RLN`
+// third is the aforementioned output buffer argument
+// rest are all other arguments to the method
+macro_rules! call_with_output_and_error_msg {
+    // this variant is needed for the case when
+    // there are zero other arguments
+    ($instance:expr, $method:ident, $error_msg:expr) => {
+        {
+            let mut output_data: Vec<u8> = Vec::new();
+            let new_instance = $instance.process();
+            if new_instance.instance.$method(&mut output_data).is_ok() {
+                let result = Uint8Array::from(&output_data[..]);
+                std::mem::forget(output_data);
+                Ok(result)
+            } else {
+                std::mem::forget(output_data);
+                Err($error_msg.into())
+            }
+        }
+    };
+    ($instance:expr, $method:ident, $error_msg:expr, $( $arg:expr ),* ) => {
+        {
+            let mut output_data: Vec<u8> = Vec::new();
+            let new_instance = $instance.process();
+            if new_instance.instance.$method($($arg.process()),*, &mut output_data).is_ok() {
+                let result = Uint8Array::from(&output_data[..]);
+                std::mem::forget(output_data);
+                Ok(result)
+            } else {
+                std::mem::forget(output_data);
+                Err($error_msg.into())
```

Comment:
like the pr for rln-wasm, can we include the error messages here too?

Reply:
Fixed here: https://github.com/vacp2p/zerokit/pull/141/commits/ce5248aa4855b2f3e2228ca1b46051db33af1915